app/assets/javascripts/surveys.js

```diff
@@ -189,14 +192,57 @@
       }
     },
 
+    processTemplate: function (args, template) {
+      $.each(args, function (key, value) {
+        template = template.replace(
+          new RegExp('\{\{' + key + '\}\}', 'g'),
+          value
+        )
+      })
+      return template
+    },
+
+    getUrlSurveyTemplate: function () {
+      return {
+        render: function(survey) {
+          var defaultUrlArgs = {
+            title: 'Tell us what you think of GOV.UK',
+            surveyCta: 'Take the 3 minute survey',
+            surveyCtaPostscript: 'This will open a short survey on another website',
+            surveyUrl: userSurveys.addCurrentPathToURL(survey.url),
+          }
+          var mergedArgs = $.extend(defaultUrlArgs, survey.templateArgs)
+          return userSurveys.processTemplate(mergedArgs, URL_SURVEY_TEMPLATE)
+        }
+      }
+    },
+
+    getEmailSurveyTemplate: function () {
+      return {
+        render: function(survey) {
+          var defaultEmailArgs = {
+            title: 'Tell us what you think of GOV.UK',
+            surveyCta: 'Take a short survey to give us your feedback',
+            surveyFormDescription: 'We’ll send you a link to a feedback form. It only takes 2 minutes to fill in.',
+            surveyFormCta: 'Send me the survey',
+            surveyFormCtaPostscript: 'Don’t worry: we won’t send you spam or share your email address with anyone.',
+            surveyFormNoEmailInvite: 'Don’t have an email address?',
+            surveySuccess: 'Thanks, we’ve sent you an email with a link to the survey.',
+            surveyFailure: 'Sorry, we’re unable to send you an email right now. Please try again later.',
+            surveyId: survey.identifier,
+            surveySource: userSurveys.currentPath(),
+            surveyUrl: userSurveys.addCurrentPathToURL(survey.url),
+          }
+          var mergedArgs = $.extend(defaultEmailArgs, survey.templateArgs)
+          return userSurveys.processTemplate(mergedArgs, EMAIL_SURVEY_TEMPLATE)
+        }
+      }
```

Comment:
This looks much nicer than the long version we had before; thanks for the suggestion @nickcolley!